lib/recursive-methods.rb

```diff
@@ -1,49 +1,134 @@
 # Authoring recursive algorithms. Add comments including time and space complexity for each method.
 
-# Time complexity: ?
-# Space complexity: ?
+# Time complexity: O(n) where n is the integer given
+# Space complexity: O(n) because the number of times the function is called is 'n' times and placed on the stack
 def factorial(n)
-    raise NotImplementedError, "Method not implemented"
+  if n < 0
+    raise ArgumentError
+  elsif n == 0
+    return 1
+  else
+    return n * factorial(n-1)
+  end
 end
 
-# Time complexity: ?
-# Space complexity: ?
+# Time complexity: O(n) where n is the length of string 's'
+# Space complexity: O(n) because the number of times the function reverse is called and placed on the stack
 def reverse(s)
```

Comment:
👍 
This works, but because you create a new array with each recursive call this is O(n<sup>2</sup>) for both time/space complexity.